tsconfig.json

```
@@ -0,0 +1,14 @@
+{
+  "include": ["**/**.js", "*.js"],
+  "exclude": ["coverage", "node_modules"],
+  "compilerOptions": {
+    "target": "es2020",
+    "lib": ["es2020"],
+    "module": "node16",
+    "checkJs": true,
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    "skipLibCheck": true,
+    "strict": true
```

```suggestion
    "customConditions": ["development"],
    "checkJs": true,
    "declaration": true,
    "emitDeclarationOnly": true,
    "exactOptionalPropertyTypes": true,
    "forceConsistentCasingInFileNames": true,
    "lib": ["es2020"],
    "module": "node16",
    "newLine": "lf",
    "strict": true,
    "target": "es2020"
```

I’m currently using these fields. Based on your recommendations for a shared config. I’ve been adding those to more places and found a few things that were needed compared to your original proposal. I’m also now adding `customConditions`, which is newer in TS, but I think good to add.

Finally, note that `skipLibCheck` might currently be needed,  in this project, for now

---

I convinced the TypeScript team that values for `forceConsistentCasingInFileNames` anf `newLine` are sane defaults. These values are the default as of version 5.0.